app/services/marketplace_service/api/marketplaces.rb

```diff
@@ -70,6 +70,62 @@ def create(params)
       return from_model(community)
     end
 
+    def set_locales(community, locales)
+      default_locale = community.locales[0]
+      community_name = community.name(default_locale)
+      locales.each { |locale| Helper.first_or_create_community_customization!(community, community_name, locale) }
+
+      settings = community.settings || {}
+      settings["locales"] = locales
+      community.settings = settings
+      community.save!
+
+      ensure_translations(community, locales).map { |translations|
+        locale = translations[:locale]
+        translations[:translations].each{ |translation|
+          add_translation(community, locale, translation[:translation_key], translation[:translation])
+        }
+      }
+      Rails.cache.delete("/translation_service/community/#{community.id}")
+    end
+
+    def ensure_translations(community, locales)
+      locales.map { |locale|
+        translations_res = TranslationService::API::Api.translations.get(community.id, {locales: [locale]})
```

Comment:
I started to think this a bit different angle: could this work in a bit different way
```
TranslationService::API::Api.translations.get(1, {locales: ['ja'], fallback_locale: 'en'}).data
  .uniq { |t| t[:translation_key] }
  .map {|t| {translation_key: t[:translation_key], translations:[{locale: 'ja', translation: t[:translation]}]}}
```
1st line:  get translations with new locale ('ja'), but use existing locale as fallback_locale
2nd line: useless line, but gives an idea how to remove duplicates if there are more locales per translation_key
3rd line:  map fetched locales in a format that can be passed to TranslationService::API::Api.translations.create(...) using new locale (i.e. 'ja')